lambdas/es/indexer/index.py

```diff
@@ -1,74 +1,220 @@
+        if text:
+            # documents will dominate memory footprint, there is also a fixed
+            # size for the rest of the doc that we do not account for
+            self.size += min(size, DOC_SIZE_LIMIT_BYTES)
+        # On types and fields, see
+        # https://www.elastic.co/guide/en/elasticsearch/reference/master/mapping.html
+        body = {
+            # Elastic native keys
+            # : is a legal character for S3 keys, so look for its last occurrence
+            # if you want to find the, potentially empty, version_id
+            "_id": f"{key}:{version_id}",
+            "_index": bucket,
+            # index will upsert (and clobber existing equivalent _ids)
+            "_op_type": "delete" if event_type == OBJECT_DELETE else "index",
+            "_type": "_doc",
+            # Quilt keys
+            # Be VERY CAREFUL changing these values as a type change can cause a
+            # mapper_parsing_exception that below code won't handle
+            "etag": etag,
+            "ext": ext,
+            "event": event_type,
+            "size": size,
+            "text": text,
+            "key": key,
+            "last_modified": last_modified.isoformat(),
+            "updated": datetime.utcnow().isoformat(),
+            "version_id": version_id
+        }
 
-def transform_meta(meta):
-    ''' Reshapes metadata for indexing in ES '''
-    helium = meta.get('helium')
-    user_meta = {}
-    comment = ''
-    target = ''
-    meta_text = ''
-    if helium:
-        user_meta = helium.pop('user_meta', {})
-        comment = helium.pop('comment', '') or ''
-        target = helium.pop('target', '') or ''
-    meta_text_parts = [comment, target]
-    if helium:
-        meta_text_parts.append(json.dumps(helium))
-    if user_meta:
-        meta_text_parts.append(json.dumps(user_meta))
-    if meta_text_parts:
-        meta_text = ' '.join(meta_text_parts)
-    result = {
-        'system_meta': helium,
-        'user_meta': user_meta,
-        'comment': comment,
-        'target': target,
-        'meta_text': meta_text
-    }
-    return result
+        body = {**body, **transform_meta(meta or {})}
+
+        body["meta_text"] = " ".join([body["meta_text"], key])
+
+        self.append_document(body)
+
+        if self.size > QUEUE_LIMIT_BYTES:
+            self.send_all()
+
+    def append_document(self, doc):
+        """append well-formed documents (used for retry or by append())"""
+        self.queue.append(doc)
+
+    def is_empty(self):
+        """is the queue empty?"""
+        return len(self.queue) == 0
+
+    def send_all(self):
+        """flush self.queue in a bulk call"""
+        if self.is_empty():
+            return
+        elastic_host = os.environ["ES_HOST"]
+        try:
+            awsauth = AWSRequestsAuth(
+                # These environment variables are automatically set by Lambda
+                aws_access_key=os.environ["AWS_ACCESS_KEY_ID"],
```

Comment:
I guess this is from the existing code, but would be nice to clean up. You're already creating a `boto3.session.Session()` below, so might as well get the credentials from it - `Session().get_credentials()`